### PR TITLE
[qmf] Use QTextDocument to parse html.

### PR DESCRIPTION
**Pull request by VDVsx**
Regular expression are not appropriated tool to parse a none regular language
like html, a proper parse should be used.
This commit introduces a dependency on QtGui making the messageserver
binary marginally bigger in size.

### CODE REVIEW

**Comment by VDVsx** [comment]
@matthewvogt @chriadam Please have a look, I could not come up with a better solution for this, RegExps are definitely not a solution here.

The binary size for ARM in debug mode grows from 4.1mb to 4.5mb, by my observations there's no visible impact on idle memory consumption or CPU peaks when this parsing is done.

**Comment by chriadam** [comment]
Probably should be done at presentation layer instead of within QMF, I guess?

**Comment by VDVsx** [comment]
Updated.

File: qmf/src/libraries/qmfclient/qmfclient.pro [hunk 1]
```diff
@@ -16,7 +16,11 @@ win32: {
     lessThan(QT_MAJOR_VERSION,5):lessThan(QT_MINOR_VERSION,6):lessThan(QT_PATCH_VERSION,2):DEFINES += QT_NO_QCOP_LOCAL_SOCKET
 }
 
-QT = core sql network
+contains(DEFINES, USE_HTML_PARSER) {
+    QT += gui
+}
+
+QT = core sql network gui
```
Comment (matthewvogt) [inline qmf/src/libraries/qmfclient/qmfclient.pro]: This looks wrong, should have the old line followed by the conditional part.

**Comment by matthewvogt** [comment]
Otherwise LGTM

**Pull request merged**